crates/ruff_python_parser/src/parser/statement.rs

```
@@ -909,6 +911,16 @@ impl<'src> Parser<'src> {
         // type x = x := 1
         let value = self.parse_conditional_expression_or_higher();
 
+        // test_ok type_stmt_py312
+        // # parse_options: {"target-version": "3.12"}
+        // type x = int
+
+        // test_err type_stmt_py311
+        // # parse_options: {"target-version": "3.11"}
+        // type x = int
+
+        self.add_unsupported_syntax_error(UnsupportedSyntaxErrorKind::TypeStmt, type_range);
+
```

I would probably move this up right after where we parsed the `type` keyword. Unless you want to suppress the error if there are any other syntax errors.

---

Oh right, I think it was down there from when I was using the whole statement range. That makes sense now!